avc/src/planner/planner.cpp

```diff
@@ -126,39 +177,107 @@ void planner::mapCb(const sensor_msgs::PointCloud2ConstPtr& map) {
     pcl::KdTreeFLANN<pcl::PointXYZ> kdtree;
     kdtree.setInputCloud(cloud);
 
-    //build paths and evaluate their weights
-    vector<double> weightedSteeringRaw;
-    double weightTotal = 0, cost, angle;
+    /* Build paths and evaluate their weights. Tracks the best weight.
+     * weightedSteerVecs holds all of the random steering samples and their
+     * path weights.
+     */
+    vector<WeightedSteeringVec> weightedSteerVecs(PATH_ITERATIONS);
+    double angle;
+    double bestWeight = 0;
     int increments = PATH_STAGE_TIME / TIME_INCREMENT;
-
     for(int i = 0; i < PATH_ITERATIONS; i++) {
         vector<double> steerPath(increments * PATH_STAGES, 0);
+        WeightedSteeringVec wsv;
         for(int s = 0; s < PATH_STAGES; s++) {
             angle = steeringSample();
+            wsv.steers.push_back(angle);
             fill_n(steerPath.begin() + s*increments, increments, angle);
         }
 
         sim_path sp = calculatePath(steerPath);
-        cost = calculatePathCost(sp, kdtree);
-        weightTotal += 1.0 / cost;
-
-        for(int j = 0; j < sp.angles.size(); j++) {
-            if(j < weightedSteeringRaw.size()) {
-                weightedSteeringRaw[j] += sp.angles[j] / cost;
-            } else {
-                weightedSteeringRaw.push_back(sp.angles[j] / cost);
+        wsv.weight = 1.0 / calculatePathCost(sp, kdtree);
+        weightedSteerVecs.push_back(wsv);
+
+        if(wsv.weight > bestWeight) bestWeight = wsv.weight;
+    }
+    //ROS_INFO("best weight %f", bestWeight);
+
+    /* Filter paths by weight and sort them by first steering value.
+     * Store the results in weightedSteerVecsFiltered.
+     */
+    vector<WeightedSteeringVec> weightedSteerVecsFiltered;
+    for(int i = 0; i < weightedSteerVecs.size(); i++) {
```

Comment:
If you're not using the index in your calculations, consider using a [range-based for loop](http://en.cppreference.com/w/cpp/language/range-for).